// nms-creative-ui/src/pages/Third Party Libraries/Extras/Chart/Preview/MixedCharts.jsx
import React, { useEffect, useRef } from "react";
import * as echarts from "echarts";
import { CanvasRenderer } from "echarts/renderers";

echarts.use(CanvasRenderer);

const MixedCharts = () => {
  const chartRef = useRef(null);
  const chartInstance = useRef(null);

  useEffect(() => {
    const initChart = () => {
      const chart = echarts.init(chartRef.current);
      chartInstance.current = chart;
      const option = {
        color: ["#00B9ED", "#FFC87C", "#FF416C"],
        title: {
          text: "Mixed Charts",
        },
        tooltip: {
          trigger: "axis",
          axisPointer: {
            type: "cross",
            label: {
              backgroundColor: "#6a7985",
            },
          },
        },
        legend: {
          data: ["Test 1", "Test 2", "Test 3"],
          icon: "circle",
          selected: {
            "Test 1": true,
            "Test 2": true,
            "Test 3": true,
          },
          orient: "horizontal",
          right: 40,
        },
        grid: {
          left: "3%",
          right: "4%",
          bottom: "3%",
          containLabel: true,
        },
        xAxis: {
          type: "category",
          data: [
            "2003",
            "Feb'03",
            "Mar'03",
            "Apr'03",
            "May'03",
            "Jun'03",
            "Jul'03",
            "Aug'03",
            "Sep'03",
            "Oct'03",
            "Nov'03",
            "Dec'03",
          ],
          boundaryGap: false,
        },
        yAxis: [
          {
            type: "value",
            min: 0,
            max: 300,
            interval: 50,
            boundaryGap: false,
          },
        ],
        series: [
          {
            name: "Test 1",
            type: "line",
            smooth: true,
            symbolSize: 10,
            type: "scatter",
            data: [100, 80, 140, 152, 90, 87, 80, 110, 95, 115, 85, 200],
          },
          {
            name: "Test 2",
            type: "line",
            stack: "Total",
            smooth: true,
            symbol: "none",
            lineStyle: {
              width: 0,
            },
            showSymbol: false,
            areaStyle: {
              opacity: 0.3,
              color: new echarts.graphic.LinearGradient(0, 0, 0, 1, [
                { offset: 0, color: "rgb(255, 200, 124)" },
                { offset: 1, color: "rgb(252, 251, 121)" },
              ]),
            },
            emphasis: {
              focus: "series",
            },
            data: [130, 92, 71, 104, 80, 110, 120, 190, 105, 132, 110, 130],
          },
          {
            name: "Test 3",
            type: "line",
            smooth: true,
            symbol: "none",
            emphasis: {
              focus: "series",
            },
            data: [40, 112, 80, 96, 83, 90, 250, 95, 70, 90, 75, 180],
          },
        ],
      };

      chart.setOption(option);
      window.addEventListener("resize", () => {
        chart.resize();
      });
    };

    initChart();

    return () => {
      window.removeEventListener("resize", () => {
        chartInstance.current.dispose();
      });
    };
  }, []);

  return (
    <div
      id="main"
      style={{ width: "100%", height: "400px" }}
      ref={chartRef}
    ></div>
  );
};

export default MixedCharts;
