// nms-creative-ui/src/components/Data Display/Badge/Badge.jsx
import React from "react";
import { Badge as NMSBadge } from "@mui/material";
import PropTypes from "prop-types";

const Badge = ({
  anchorOrigin,
  badgeContent,
  children,
  color,
  component,
  invisible,
  max,
  overlap,
  sx,
  variant,
  ...props
}) => {
  return (
    <NMSBadge
      anchorOrigin={anchorOrigin}
      badgeContent={badgeContent}
      color={color}
      component={component}
      invisible={invisible}
      max={max}
      overlap={overlap}
      sx={sx}
      variant={variant}
      {...props}
    >
      {children}
    </NMSBadge>
  );
};

Badge.propTypes = {
  anchorOrigin: PropTypes.shape({
    horizontal: PropTypes.oneOf(["left", "right"]).isRequired,
    vertical: PropTypes.oneOf(["bottom", "top"]).isRequired,
  }),
  badgeContent: PropTypes.node,
  children: PropTypes.node,
  color: PropTypes.oneOfType([
    PropTypes.oneOf([
      "default",
      "primary",
      "secondary",
      "error",
      "info",
      "success",
      "warning",
    ]),
    PropTypes.string,
  ]),
  component: PropTypes.elementType,
  invisible: PropTypes.bool,
  max: PropTypes.number,
  overlap: PropTypes.oneOf(["circular", "rectangular"]),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["dot", "standard"]),
    PropTypes.string,
  ]),
  ...NMSBadge.PropTypes,
};

Badge.defaultProps = {
  children: "NMS Badge",
  color: "primary",
  max: 99,
  overlap: "rectangular",
  variant: "standard",
};

export default Badge;
