// nms-creative-ui/src/components/Input/Text Field/TextField.jsx
import React from "react";
import { TextField as NMSTextField, InputAdornment } from "@mui/material";
import PropTypes from "prop-types";

const TextField = ({
  id,
  helperText,
  variant,
  startAdornment,
  endAdornment,
  multiline,
  fullWidth,
  ...props
}) => {
  return (
    <NMSTextField
      id={id}
      fullWidth={fullWidth}
      helperText={helperText}
      variant={variant}
      multiline={multiline}
      {...props}
      InputProps={{
        startAdornment: startAdornment && (
          <InputAdornment position="start">{startAdornment}</InputAdornment>
        ),
        endAdornment: endAdornment && (
          <InputAdornment position="end">{endAdornment}</InputAdornment>
        ),
      }}
    />
  );
};

TextField.propTypes = {
  disabled: PropTypes.bool,
  endAdornment: PropTypes.node,
  fullWidth: PropTypes.bool,
  helperText: PropTypes.string,
  label: PropTypes.string,
  size: PropTypes.string,
  startAdornment: PropTypes.node,
  sx: PropTypes.arrayOf(PropTypes.func),
  variant: PropTypes.string,
};

TextField.defaultProps = {
  color: "primary",
  size: "medium",
  variant: "outlined",
};

export default TextField;
