// nms-creative-ui/src/pages/Components/Feedback/Dialog/Preview/FullScreenDialog.jsx
import React from "react";
import { Fragment } from "react";
import {
  Button,
  Dialog,
  DialogContent,
  DialogTitle,
} from "../../../../../components";
import { useState } from "react";
import { Slide } from "@mui/material";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const FullScreenDialog = () => {
  const [open, setOpen] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleCloseBasic = (value) => {
    setOpen(false);
  };
  return (
    <Fragment>
      <Button onClick={handleClickOpen}>Full Screen Dialog</Button>

      <Dialog
        fullWidth
        fullScreen
        open={open}
        onClose={handleCloseBasic}
        TransitionComponent={Transition}
        PaperProps={{
          style: {
            margin: 0,
            maxWidth: "100%",
            width: "100%",
            height: "100%",
            borderRadius: "0px",
          },
        }}
      >
        <DialogTitle>Full Screen Dialog</DialogTitle>
        <DialogContent>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec
          nunc ipsum. Vestibulum id mauris mi. Aliquam molestie nunc non erat
          cursus bibendum. Nunc ac erat tellus. Nam aliquam ac sapien vel
          dictum. Integer at tortor ac urna interdum aliquam. In sit amet
          placerat nisl, sed rutrum nunc. Proin porttitor vel arcu eu vulputate.
          Vestibulum a porta lacus. Cras pharetra ex et orci feugiat, at congue
          metus pharetra. Donec imperdiet ullamcorper vulputate. Donec elementum
          accumsan justo, et rhoncus diam aliquam eget. Vestibulum sit amet
          lectus a nibh dignissim sollicitudin in egestas urna. Sed quam sapien,
          hendrerit eget auctor sed, dictum malesuada turpis. Mauris tempor
          dolor lectus, in mollis nulla varius sit amet. Donec bibendum diam non
          lacus auctor viverra. Praesent eu massa id mi consequat varius
          condimentum at enim. Sed tempor lectus placerat odio vulputate, vel
          malesuada mauris blandit. Nullam ut consequat massa. Quisque neque
          urna, euismod sed aliquam non, ultricies commodo neque. Suspendisse et
          varius ipsum. Vestibulum rhoncus condimentum tellus, eget consectetur
          ante egestas non. Curabitur porttitor interdum ultricies.
        </DialogContent>
      </Dialog>
    </Fragment>
  );
};

export default FullScreenDialog;
