// nms-creative-ui/src/pages/Components/Surfaces/Card/Preview/ClickableCard.jsx
import React from "react";
import { Typography } from "@mui/material";
import {
  Box,
  Card,
  CardContent,
  Grid,
  Stack,
} from "../../../../../components";
import { Link } from "react-router-dom";
import * as SolarIconSet from "solar-icon-set";

const ClickableCard = () => {
  const DATA = [
    {
      id: 0,
      bgImage: "/banners/banner1.png",
      desc: "Lorem ipsum dolor sit amet amet",
      title: "Title",
      icon: (
        <SolarIconSet.UsersGroupTwoRounded
          size={97}
          iconStyle="BoldDuotone"
          color="#FFFFFF"
        />
      ),
    },
    {
      id: 1,
      bgImage: "/banners/banner2.png",
      desc: "Lorem ipsum dolor sit amet amet",
      title: "Title",
      icon: (
        <SolarIconSet.UsersGroupTwoRounded
          size={97}
          iconStyle="BoldDuotone"
          color="#FFFFFF"
        />
      ),
    },
    {
      id: 2,
      bgImage: "/banners/banner3.png",
      desc: "Lorem ipsum dolor sit amet amet",
      title: "Title",
      icon: (
        <SolarIconSet.UsersGroupTwoRounded
          size={97}
          iconStyle="BoldDuotone"
          color="#FFFFFF"
        />
      ),
    },
  ];

  return (
    <Grid container spacing={2}>
      {DATA.map((item, index) => (
        <Grid key={index} item mobile={12} desktop={6}>
          <Link
            style={{
              textDecoration: "none",
              color: "none",
            }}
          >
            <Card
              sx={{
                backgroundSize: "cover",
                backgroundImage: `url(${item.bgImage})`,
                padding: "24px",
              }}
            >
              <Box
                sx={{
                  display: "flex",
                  padding: "24px",
                }}
              >
                <CardContent>
                  <Stack
                    direction="row"
                    spacing={4}
                    justifyContent="space-between"
                  >
                    <Box sx={{ display: "flex", alignItems: "center" }}>
                      <Stack direction="column">
                        <Typography color="#FFFFFF" variant="body1">
                          {item.desc}
                        </Typography>
                        <Typography color="#FFFFFF" variant="h3">
                          {item.title}
                        </Typography>
                      </Stack>
                    </Box>
                    {item.icon}
                  </Stack>
                </CardContent>
              </Box>
            </Card>
          </Link>
        </Grid>
      ))}
    </Grid>
  );
};

export default ClickableCard;
