// nms-creative-ui/src/components/Surfaces/Accordion/Accordion/Accordion.jsx
import React from "react";
import { Accordion as NMSAccordion } from "@mui/material";
import PropTypes from "prop-types";

const Accordion = ({
  children,
  expanded,
  onChange,
  slotProps,
  slots,
  sx,
  ...props
}) => {
  return (
    <NMSAccordion
      expanded={expanded}
      onChange={onChange}
      slotProps={slotProps}
      slots={slots}
      sx={sx}
      {...props}
    >
      {children}
    </NMSAccordion>
  );
};

Accordion.propTypes = {
  children: PropTypes.node.isRequired,
  expanded: PropTypes.bool,
  onChange: PropTypes.func,
  slotProps: PropTypes.shape({
    transition: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
  }),
  slots: PropTypes.shape({
    transition: PropTypes.elementType,
  }),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  ...NMSAccordion.PropTypes,
};

Accordion.defaultProps = {
  children: "Accordion"
}

export default Accordion;
