// nms-creative-ui/src/components/Navigation/Pagination/PaginationItem/PaginationItem.jsx
import React from "react";
import { PaginationItem as NMSPaginationItem } from "@mui/material";
import PropTypes from "prop-types";

const PaginationItem = ({
  color,
  component,
  page,
  shape,
  size,
  slots,
  sx,
  type,
  variant,
  ...props
}) => {
  return (
    <NMSPaginationItem
      color={color}
      component={component}
      page={page}
      shape={shape}
      size={size}
      slots={slots}
      sx={sx}
      type={type}
      variant={variant}
      {...props}
    />
  );
};

PaginationItem.propTypes = {
  color: PropTypes.oneOfType([
    PropTypes.oneOf(["primary", "secondary", "standard"]),
    PropTypes.string,
  ]),
  component: PropTypes.elementType,
  page: PropTypes.node,
  shape: PropTypes.oneOf(["circular", "rounded"]),
  size: PropTypes.oneOfType([
    PropTypes.oneOf(["small", "medium", "large"]),
    PropTypes.string,
  ]),
  slots: PropTypes.shape({
    first: PropTypes.elementType,
    last: PropTypes.elementType,
    next: PropTypes.elementType,
    previous: PropTypes.elementType,
  }),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  type: PropTypes.oneOfType([
    PropTypes.oneOf([
      "end-ellipsis",
      "first",
      "last",
      "next",
      "page",
      "previous",
      "start-ellipsis",
    ]),
    PropTypes.string,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["outlined", "text"]),
    PropTypes.string,
  ]),
  ...NMSPaginationItem.PropTypes,
};

PaginationItem.defaultProps = {
  color: "standard",
  shape: "circular",
  size: "medium",
  type: "page",
  variant: "text",
};

export default PaginationItem;
