// nms-creative-ui/src/components/Input/Button/ButtonGroup/ButtonGroup.jsx
import React from "react";
import { ButtonGroup as NMSButtonGroup } from "@mui/material";
import PropTypes from "prop-types";

const ButtonGroup = ({
  children,
  color,
  component,
  orientation,
  size,
  sx,
  variant,
  ...props
}) => {
  return (
    <NMSButtonGroup
      color={color}
      component={component}
      orientation={orientation}
      size={size}
      sx={sx}
      variant={variant}
    >
      {children}
    </NMSButtonGroup>
  );
};

ButtonGroup.propTypes = {
  children: PropTypes.node,
  color: PropTypes.oneOfType([
    PropTypes.oneOf([
      "inherit",
      "primary",
      "secondary",
      "error",
      "info",
      "success",
      "warning",
    ]),
    PropTypes.string,
  ]),
  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
  size: PropTypes.oneOfType([
    PropTypes.oneOf(["small", "medium", "large"]),
    PropTypes.string,
  ]),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["contained", "outlined", "text"]),
    PropTypes.string,
  ]),
};

ButtonGroup.defaultProps = {
  children: "NMS ButtonGroup",
  color: "primary",
  orientation: "horizontal",
  size: "medium",
  variant: "outlined",
};

export default ButtonGroup;
