// nms-creative-ui/src/components/MUI X/Date Time Picker/Time/TimeClock/TimeClock.jsx
import React from "react";
import { TimeClock as NMSTimeClock } from "@mui/x-date-pickers";
import PropTypes from "prop-types";

const TimeClock = ({
  defaultValue,
  onChange,
  openTo,
  referenceDate,
  slotProps,
  slots,
  sx,
  timeStep,
  timezone,
  value,
  view,
  views,
  ...props
}) => {
  return (
    <NMSTimeClock
      defaultValue={defaultValue}
      onChange={onChange}
      openTo={openTo}
      referenceDate={referenceDate}
      slotProps={slotProps}
      slots={slots}
      sx={sx}
      timeStep={timeStep}
      timezone={timezone}
      value={value}
      view={view}
      views={views}
      {...props}
    />
  );
};

TimeClock.propTypes = {
  defaultValue: PropTypes.any,
  onChange: PropTypes.func,
  openTo: PropTypes.oneOf(["hours", "minutes", "seconds"]),
  referenceDate: PropTypes.any,
  slotProps: PropTypes.object,
  slots: PropTypes.object,
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  timeStep: PropTypes.shape({
    hours: PropTypes.number,
    minutes: PropTypes.number,
    seconds: PropTypes.number,
  }),
  timezone: PropTypes.string,
  value: PropTypes.any,
  view: PropTypes.oneOf(["hours", "minutes", "seconds"]),
  views: PropTypes.arrayOf(["hours", "minutes", "seconds"]),
  ...NMSTimeClock.propTypes,
};

export default TimeClock;
