// nms-creative-ui/src/components/Foundation/Typography/Typography.jsx
import React from "react";
import { Typography as NMSTypography } from "@mui/material";
import PropTypes from "prop-types";

const Typography = ({
  align,
  children,
  component,
  sx,
  variant,
  variantMapping,
  disabled,
  ...props
}) => {
  return (
    <NMSTypography
      align={align}
      component={component}
      sx={sx}
      variant={variant}
      variantMapping={variantMapping}
      disabled={disabled}
      {...props}
    >
      {children}
    </NMSTypography>
  );
};

Typography.propTypes = {
  align: PropTypes.oneOf(["center", "inherit", "justify", "left", "right"]),
  children: PropTypes.node,
  component: PropTypes.elementType,
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf([
      "body1",
      "body2",
      "button",
      "caption",
      "h1",
      "h2",
      "h3",
      "h4",
      "h5",
      "h6",
      "inherit",
      "overline",
      "subtitle1",
      "subtitle2",
      "buttonLarge",
      "buttonMedium",
      "buttonSmall",
    ]),
    PropTypes.string,
  ]),
  variantMapping: PropTypes.object,
};

Typography.defaultProps = {
  align: "inherit",
  variant: "body1",
  variantMapping: {
    h1: "h1",
    h2: "h2",
    h3: "h3",
    h4: "h4",
    h5: "h5",
    h6: "h6",
    subtitle1: "h6",
    subtitle2: "h6",
    body1: "p",
    body2: "p",
    inherit: "p",
    buttonLarge: "buttonLarge",
    buttonMedium: "buttonMedium",
    buttonSmall: "buttonSmall"
  },
};

export default Typography;
