// nms-creative-ui/src/pages/Components/Navigation/Menu/JSX/Customization.jsx
export const CustomizationExample = `import React, { Fragment } from 'react'
import * as SolarIconSet from "solar-icon-set";
import { Button, Icon, Menu, MenuItem } from 'nms-creative-ui';
import { Divider, styled, alpha } from '@mui/material';


const StyledMenu = styled((props) => (
  <Menu
    elevation={0}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'right',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'right',
    }}
    {...props}
  />
))(({ theme }) => ({
  '& .MuiPaper-root': {
    borderRadius: 6,
    marginTop: theme.spacing(1),
    minWidth: 180,
    color:
      theme.palette.mode === 'light' ? 'rgb(55, 65, 81)' : theme.palette.grey[300],
    boxShadow:
      'rgb(255, 255, 255) 0px 0px 0px 0px, rgba(0, 0, 0, 0.05) 0px 0px 0px 1px, rgba(0, 0, 0, 0.1) 0px 10px 15px -3px, rgba(0, 0, 0, 0.05) 0px 4px 6px -2px',
    '& .MuiMenu-list': {
      padding: '4px 0',
    },
    '& .MuiMenuItem-root': {
      '& .MuiSvgIcon-root': {
        fontSize: 18,
        color: theme.palette.text.secondary,
        marginRight: theme.spacing(1.5),
      },
      '&:active': {
        backgroundColor: alpha(
          theme.palette.primary.main,
          theme.palette.action.selectedOpacity,
        ),
      },
    },
  },
}));

const Customization = () => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);
  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <Fragment>
      <Button
        id="demo-customized-button"
        aria-controls={open ? 'demo-customized-menu' : undefined}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        variant="contained"
        disableElevation
        onClick={handleClick}
        endIcon={<Icon solariconset={SolarIconSet.AltArrowDown} size={24} iconStyle={"BoldDuotone"} sx={{
          display: "flex",
          justifyContent: "center",
        }}/>}
      >
        Options
      </Button>
      <StyledMenu
        id="demo-customized-menu"
        MenuListProps={{
          'aria-labelledby': 'demo-customized-button',
        }}
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
      >
        <MenuItem onClick={handleClose} disableRipple>
        <Icon solariconset={SolarIconSet.Pen} size={24} iconStyle={"BoldDuotone"}/>
          Edit
        </MenuItem>
        <MenuItem onClick={handleClose} disableRipple>
        <Icon solariconset={SolarIconSet.Copy} size={24} iconStyle={"BoldDuotone"}/>
          Duplicate
        </MenuItem>
        <Divider sx={{ my: 0.5 }} />
        <MenuItem onClick={handleClose} disableRipple>
        <Icon solariconset={SolarIconSet.ArchiveMinimalistic} size={24} iconStyle={"BoldDuotone"}/>
          Archive
        </MenuItem>
        <MenuItem onClick={handleClose} disableRipple>
        <Icon solariconset={SolarIconSet.MenuDots} size={24} iconStyle={"BoldDuotone"}/>
          More
        </MenuItem>
      </StyledMenu>
    </Fragment>
  );
}

export default Customization;`
