// nms-creative-ui/src/components/Navigation/Tabs/TabScrollButton/TabScrollButton.jsx
import React from "react";
import { TabScrollButton as NMSTabScrollButton } from "@mui/material";
import PropTypes from "prop-types";

const TabScrollButton = ({
  direction,
  orientation,
  children,
  slots,
  sx,
  ...props
}) => {
  return (
    <NMSTabScrollButton
      direction={direction}
      orientation={orientation}
      slots={slots}
      sx={sx}
      {...props}
    >
      {children}
    </NMSTabScrollButton>
  );
};

TabScrollButton.propTypes = {
  direction: PropTypes.oneOf(["left", "right"]),
  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
  children: PropTypes.node,
  slots: PropTypes.shape({
    EndScrollButtonIcon: PropTypes.elementType,
    StartScrollButtonIcon: PropTypes.elementType,
  }),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  ...NMSTabScrollButton.PropTypes,
};

export default TabScrollButton;
