// nms-creative-ui/src/pages/Components/Inputs/Button/JSX/Contained/CustomizedExample.jsx
export const CustomizedExample = `import React from "react";
import { Button } from "nms-creative-ui";
import { Typography, styled } from "@mui/material";

const Customized = () => {
  const NMSCustomizedButton = styled(Button)({
    boxShadow: "none",
    backgroundColor: "#fff",
    color: "#FF4545",
    "&:hover":{
      backgroundColor: "#DB3241",
      color: "#fff",
    },
    "*.MuiTypography-root": {
      textTransform: "none",
      fontSize: 18,
      fontWeight: 500,
      padding: "6px 12px",
      fontFamily: [
        "-apple-system",
        "BlinkMacSystemFont",
        '"Segoe UI"',
        "Roboto",
        '"Helvetica Neue"',
        "Arial",
        "sans-serif",
        '"Apple Color Emoji"',
        '"Segoe UI Emoji"',
        '"Segoe UI Symbol"',
      ],
    },
  });

  return (
    <NMSCustomizedButton
      variant="contained"
      disableRipple
      startIcon={
        <img
          src="/nmsLogo.svg"
          alt="NMS LOGO"
          sx={{
            height: "31px",
            width: "50px",
          }}
        />
      }
    >
      <Typography>Customized</Typography>
    </NMSCustomizedButton>
  );
};

export default Customized;`;
