// nms-creative-ui/src/components/Generic Components/Sidebar Components/DesktopDrawer.jsx
import { CssBaseline, styled } from "@mui/material";
import React, { Fragment, useState } from "react";
import * as SolarIconSet from "solar-icon-set";
import { Drawer } from "../General Components/CustDrawer";
import { Box, Icon, IconButton } from "../..";
import SideBarNavHelpers from "./SideBarNavHelpers";


const drawerWidth = 248;
const DrawerHeader = styled("div")(({ theme }) => ({
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  }));

const DesktopDrawer = ({ open, handleDrawerClose, handleDrawerOpen }) => {
  return (
    <Drawer
      variant="permanent"
      open={open}
      sx={{
        "& .MuiDrawer-paper": {
          boxSizing: "border-box",
          width: drawerWidth,
        },
      }}
    >
      <Box>
        <DrawerHeader>
          {open && (
            <IconButton onClick={handleDrawerClose}>
              <Icon color="secondary">
                <SolarIconSet.AltArrowLeft size={22} iconStyle="Outline" />
              </Icon>
            </IconButton>
          )}

          {!open && (
            <IconButton onClick={handleDrawerOpen}>
              <Icon color="secondary">
                <SolarIconSet.AltArrowRight size={22} iconStyle="Outline" />
              </Icon>
            </IconButton>
          )}
        </DrawerHeader>
        <SideBarNavHelpers open={open} />
      </Box>
    </Drawer>
  );
};

export default DesktopDrawer;
