// nms-creative-ui/src/components/Generic Components/Sidebar Components/NavItem.jsx
import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import ListItem from "../../Data Display/List/ListItem";
import ListItemIcon from "../../Data Display/List/ListItemIcon";
import ListItemText from "../../Data Display/List/ListItemText";
import Typography from "../../Foundation/Typography";
import Icon from "../../Foundation/Icon";
import CircleIcon from "@mui/icons-material/Circle";

const NavItem = ({ key, path, activeItem, icon, title, open }) => {
  const truncate = (input) =>
    input?.length > 12 ? `${input.substring(0, 12)}...` : input;

  return (
    <ListItem
      key={key}
      component={Link}
      to={path}
      selected={activeItem}
      sx={{
        padding: "10px",
        ...(open
          ? {
              display: "flex",
              flexDirection: "column",
              alignItems: "center",
              borderRadius: "6px",
            }
          : {
              gap: "10px",
            }),
      }}
    >
      {icon ? (
        <ListItemIcon
          sx={{
            ...(!open
              ? { minWidth: "0" }
              : { display: "flex", justifyContent: "center" }),
          }}
        >
          {icon}
        </ListItemIcon>
      ) : (
        <ListItemIcon
          sx={{
            ...(!open
              ? { minWidth: "0" }
              : { display: "flex", justifyContent: "center" }),
          }}
        >
          <Icon
            sx={{
              display: "flex",
              alignItems: "center",
              justifyContent: "center",
            }}
          >
            <CircleIcon sx={{ fontSize: "6px" }} />
          </Icon>
        </ListItemIcon>
      )}
      <ListItemText
        primary={
          open ? (
            <Typography variant={"small"}>{truncate(title)}</Typography>
          ) : (
            <Typography noWrap>{title}</Typography>
          )
        }
        sx={{ color: "#454F5B" }}
      />
    </ListItem>
  );
};

export default NavItem;
