// nms-creative-ui/src/components/Generic Components/Sidebar Components/SubmenuItem.jsx
import React from "react";
import ListItem from "../../Data Display/List/ListItem";
import ListItemIcon from "../../Data Display/List/ListItemIcon";
import ListItemText from "../../Data Display/List/ListItemText";
import { ExpandLess, ExpandMore } from "@mui/icons-material";
import Typography from "../../Foundation/Typography";

const SubmenuItem = ({
  item,
  open,
  handleClick,
  openSections,
  useDropdown,
  activeItem,
}) => {
  const truncate = (input) =>
    input?.length > 12 ? `${input.substring(0, 12)}...` : input;
  return (
    <>
      <ListItem
        selected={activeItem}
        onClick={() => handleClick(item.submenu)}
        sx={{
          minHeight: 48,
          justifyContent: open ? "initial" : "center",
          alignItems: open ? "initial" : "center",
          padding: "10px",
          gap: "10px",
          display: useDropdown ? null : "flex",
          flexDirection: useDropdown ? null : "column",
        }}
      >
        {useDropdown ? (
          <>
            <ListItemIcon
              sx={{
                minWidth: "0",
              }}
            >
              {item.icon}
            </ListItemIcon>
            <ListItemText primary={item.title} />
            {openSections[item.submenu] ? <ExpandLess /> : <ExpandMore />}
          </>
        ) : (
          <>
            <ListItemIcon
              sx={{
                display: "flex",
                justifyContent: "center",
              }}
            >
              {item.icon}
            </ListItemIcon>
            <ListItemText
              primary={
                <Typography variant={"small"}>
                  {truncate(item.title)}
                </Typography>
              }
            />
          </>
        )}
      </ListItem>
    </>
  );
};

export default SubmenuItem;
