// nms-creative-ui/src/config/components.jsx
import { lazy } from "react";

const UnderConstruction = lazy(
  () => import("../pages/CustomPage/UnderConstruction")
);

const NotAvailable = lazy(
  () => import("../pages/CustomPage/NotAvailable")
);
const BoxPage = lazy(() => import("../pages/Components/Layout/Box/BoxPage"));
const ContainerPage = lazy(
  () => import("../pages/Components/Layout/Container/ContainerPage")
);
const StackPage = lazy(
  () => import("../pages/Components/Layout/Stack/StackPage")
);
const PaperPage = lazy(
  () => import("../pages/Components/Surfaces/Paper/PaperPage")
);
// const HomePage = lazy(() => import("../pages/Home/HomePage"));
const CardPage = lazy(
  () => import("../pages/Components/Surfaces/Card/CardPage")
);
const ChartPage = lazy(
  () => import("../pages/Third Party Libraries/Extras/Chart/ChartPage")
);
const RichTextPage = lazy(
  () => import("../pages/Third Party Libraries/Extras/RichText/RichTextPage")
);
const UploadFilePage = lazy(
  () =>
    import("../pages/Third Party Libraries/Extras/UploadFile/UploadFilePage")
);
const FormValidationPage = lazy(
  () => import("../pages/Third Party Libraries/Extras/FormValidation/FormValidationPage")
);
const DialogPage = lazy(
  () => import("../pages/Components/Feedback/Dialog/DialogPage")
);
const ProgressPage = lazy(
  () => import("../pages/Components/Feedback/Progress/ProgressPage")
);
const SnackbarPage = lazy(
  () => import("../pages/Components/Feedback/Snackbar/SnackbarPage")
);
const AccordionPage = lazy(
  () => import("../pages/Components/Surfaces/Accordion/AccordionPage")
);
const MenuPage = lazy(
  () => import("../pages/Components/Navigation/Menu/MenuPage")
);
const PaginationPage = lazy(
  () => import("../pages/Components/Navigation/Pagination/PaginationPage")
);
const TabsPage = lazy(
  () => import("../pages/Components/Navigation/Tabs/TabsPage")
);
const DataGridPage = lazy(
  () => import("../pages/Components/MUI X/DataGrid/DataGridPage")
);
const DateTimePickerPage = lazy(
  () => import("../pages/Components/MUI X/DateTimePicker/DateTimePickerPage")
);
const AvatarPage = lazy(
  () => import("../pages/Components/Data Display/Avatar/AvatarPage")
);
const BadgePage = lazy(
  () => import("../pages/Components/Data Display/Badge/BadgePage")
);
const ChipPage = lazy(
  () => import("../pages/Components/Data Display/Chip/ChipPage")
);
const ListPage = lazy(
  () => import("../pages/Components/Data Display/List/ListPage")
);
const AutocompletePage = lazy(
  () => import("../pages/Components/Inputs/Autocomplete/AutocompletePage")
);
const ButtonPage = lazy(
  () => import("../pages/Components/Inputs/Button/ButtonPage")
);
const CheckboxPage = lazy(
  () => import("../pages/Components/Inputs/Checkbox/CheckboxPage")
);
const RadioGroupPage = lazy(
  () => import("../pages/Components/Inputs/RadioGroup/RadioGroupPage")
);
const SelectPage = lazy(
  () => import("../pages/Components/Inputs/Select/SelectPage")
);
const SwitchPage = lazy(
  () => import("../pages/Components/Inputs/Switch/SwitchPage")
);
const TextFieldPage = lazy(
  () => import("../pages/Components/Inputs/Textfield/TextFieldPage")
);
const IntroductionPage = lazy(
  () => import("../pages/Getting Started/IntroductionPage")
);
const ColorsPage = lazy(
  () => import("../pages/Getting Started/Foundation Page/Colors/ColorsPage")
);
const GridPage = lazy(
  () => import("../pages/Getting Started/Foundation Page/Grid/GridPage")
);
const IconsPage = lazy(
  () => import("../pages/Getting Started/Foundation Page/Icons/IconsPage")
);
const LayoutPage = lazy(
  () => import("../pages/Getting Started/Foundation Page/Layout/LayoutPage")
);
const ShadowsPage = lazy(
  () => import("../pages/Getting Started/Foundation Page/Shadow/ShadowsPage")
);
const TypographyPage = lazy(
  () =>
    import("../pages/Getting Started/Foundation Page/Typography/TypographyPage")
);
const InstallationPage = lazy(
  () => import("../pages/Getting Started/InstallationPage")
);
const SupportedPlatformsPage = lazy(
  () => import("../pages/Getting Started/SupportedPlatformsPage")
);
const TemplatesPage = lazy(
  () => import("../pages/Getting Started/TemplatesPage")
);
const UsagePage = lazy(() => import("../pages/Getting Started/UsagePage"));

export const COMPONENTS = {
  home: <IntroductionPage />,
  error_page: {
    notAvailable: <NotAvailable/>,
  },
  getting_started: {
    introduction: <IntroductionPage />,
    installation: <InstallationPage />,
    usage: <UsagePage />,
    templates: <TemplatesPage />,
    supported_platforms: <SupportedPlatformsPage />,
    foundation: {
      layout: <LayoutPage />,
      colors: <ColorsPage />,
      typography: <TypographyPage />,
      shadows: <ShadowsPage />,
      grid: <GridPage />,
      icons: <IconsPage />,
    },
  },
  components: {
    inputs: {
      autocomplete: <AutocompletePage />,
      button: <ButtonPage />,
      checkbox: <CheckboxPage />,
      radio_group: <RadioGroupPage />,
      select: <SelectPage />,
      switch: <SwitchPage />,
      text_field: <TextFieldPage />,
    },
    data_display: {
      avatar: <AvatarPage />,
      badge: <BadgePage />,
      chip: <ChipPage />,
      list: <ListPage />,
    },
    feedback: {
      dialog: <DialogPage />,
      progress: <ProgressPage />,
      snackbar: <SnackbarPage />,
    },
    surfaces: {
      accordion: <AccordionPage />,
      card: <CardPage />,
      paper: <PaperPage />,
    },
    navigation: {
      menu: <MenuPage />,
      pagination: <PaginationPage />,
      tabs: <TabsPage />,
    },
    layout: {
      box: <BoxPage />,
      container: <ContainerPage />,
      stack: <StackPage />,
    },
    mui_x: {
      data_grid: <DataGridPage />,
      date_time_picker: <DateTimePickerPage />,
    },
  },
  third_party_libraries: {
    extras: {
      chart: <ChartPage />,
      rich_text: <RichTextPage />,
      upload_file: <UploadFilePage />,
      form_validation: <FormValidationPage />,
    },
  },
};
