// nms-creative-ui/src/config/pageList.jsx
import { COMPONENTS } from "./components";
import { SECTIONS, SUBMENU } from "./enums";
import { NAVICONS } from "./icons";

export const PAGELIST = [
  // {
  //   title: "Introduction",
  //   name: "introduction",
  //   component: COMPONENTS.getting_started.introduction,
  //   icon: NAVICONS.getting_started.introduction,
  //   key: "",
  //   path: "/",
  //   section: SECTIONS.GETTING_STARTED,
  //   submenu: "",
  // },
  {
    title: "Introduction",
    name: "introduction",
    component: COMPONENTS.getting_started.introduction,
    icon: NAVICONS.getting_started.introduction,
    key: "",
    path: "/",
    section: SECTIONS.GETTING_STARTED,
    submenu: "",
  },
  {
    title: "NotAvailable",
    component: COMPONENTS.error_page.notAvailable,
    path: "/notAvailable",
  },
  {
    title: "Installation",
    name: "installation",
    component: COMPONENTS.getting_started.installation,
    icon: NAVICONS.getting_started.installation,
    key: "",
    path: "/getting-started/installation",
    section: SECTIONS.GETTING_STARTED,
    submenu: "",
  },
  {
    title: "Usage",
    name: "usage",
    component: COMPONENTS.getting_started.usage,
    icon: NAVICONS.getting_started.usage,
    key: "",
    path: "/getting-started/usage",
    section: SECTIONS.GETTING_STARTED,
    submenu: "",
  },
  {
    title: "Templates",
    name: "templates",
    component: COMPONENTS.getting_started.templates,
    icon: NAVICONS.getting_started.templates,
    key: "",
    path: "/getting-started/templates",
    section: SECTIONS.GETTING_STARTED,
    submenu: "",
  },
  {
    title: "Supported Platforms",
    name: "supported_platforms",
    component: COMPONENTS.getting_started.supported_platforms,
    icon: NAVICONS.getting_started.supported_platforms,
    key: "",
    path: "/getting-started/supported-platforms",
    section: SECTIONS.GETTING_STARTED,
    submenu: "",
  },
  {
    title: "Foundation",
    name: "foundation",
    component: COMPONENTS.getting_started.foundation,
    icon: NAVICONS.getting_started.foundation,
    key: "",
    path: "",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: true,
    groupBy: "foundation",
  },
  {
    title: "Layout",
    name: "layout",
    component: COMPONENTS.getting_started.foundation.layout,
    icon: "",
    key: "",
    path: "/getting-started/foundation/layout",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Colors",
    name: "colors",
    component: COMPONENTS.getting_started.foundation.colors,
    icon: "",
    key: "",
    path: "/getting-started/foundation/colors",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Typography",
    name: "typography",
    component: COMPONENTS.getting_started.foundation.typography,
    icon: "",
    key: "",
    path: "/getting-started/foundation/typography",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Shadows",
    name: "shadows",
    component: COMPONENTS.getting_started.foundation.shadows,
    icon: "",
    key: "",
    path: "/getting-started/foundation/shadows",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Grid",
    name: "grid",
    component: COMPONENTS.getting_started.foundation.grid,
    icon: "",
    key: "",
    path: "/getting-started/foundation/grid",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Icons",
    name: "icons",
    component: COMPONENTS.getting_started.foundation.icons,
    icon: "",
    key: "",
    path: "/getting-started/foundation/icons",
    section: SECTIONS.GETTING_STARTED,
    submenu: SUBMENU.FOUNDATION,
    isMain: false,
    groupBy: "foundation",
  },
  {
    title: "Inputs",
    name: "inputs",
    component: COMPONENTS.components.inputs,
    icon: NAVICONS.components.inputs,
    key: "",
    path: "",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: true,
    groupBy: "inputs",
  },
  {
    title: "Autocomplete",
    name: "autocomplete",
    component: COMPONENTS.components.inputs.autocomplete,
    icon: "",
    key: "",
    path: "/components/inputs/autocomplete",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Button",
    name: "button",
    component: COMPONENTS.components.inputs.button,
    icon: "",
    path: "/components/inputs/button",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Checkbox",
    name: "checkbox",
    component: COMPONENTS.components.inputs.checkbox,
    icon: "",
    key: "",
    path: "/components/inputs/checkbox",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Radio Group",
    name: "radio_group",
    component: COMPONENTS.components.inputs.radio_group,
    icon: "",
    key: "",
    path: "/components/inputs/radio-group",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Select",
    name: "select",
    component: COMPONENTS.components.inputs.select,
    icon: "",
    key: "",
    path: "/components/inputs/select",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Switch",
    name: "switch",
    component: COMPONENTS.components.inputs.switch,
    icon: "",
    key: "",
    path: "/components/inputs/switch",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Text Field",
    name: "text_field",
    component: COMPONENTS.components.inputs.text_field,
    icon: "",
    key: "",
    path: "/components/inputs/text-field",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.INPUTS,
    isMain: false,
    groupBy: "inputs",
  },
  {
    title: "Data Display",
    name: "data_display",
    component: COMPONENTS.components.data_display,
    icon: NAVICONS.components.data_display,
    key: "",
    path: "/components/data-display/",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.DATA_DISPLAY,
    isMain: true,
    groupBy: "data_display",
  },
  {
    title: "Avatar",
    name: "avatar",
    component: COMPONENTS.components.data_display.avatar,
    icon: "",
    key: "",
    path: "/components/data-display/avatar",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.DATA_DISPLAY,
    isMain: false,
    groupBy: "data_display",
  },
  {
    title: "Badge",
    name: "badge",
    component: COMPONENTS.components.data_display.badge,
    icon: "",
    key: "",
    path: "/components/data-display/badge",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.DATA_DISPLAY,
    isMain: false,
    groupBy: "data_display",
  },
  {
    title: "Chip",
    name: "chip",
    component: COMPONENTS.components.data_display.chip,
    icon: "",
    key: "",
    path: "/components/data-display/chip",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.DATA_DISPLAY,
    isMain: false,
    groupBy: "data_display",
  },
  {
    title: "List",
    name: "list",
    component: COMPONENTS.components.data_display.list,
    icon: "",
    key: "",
    path: "/components/data-display/list",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.DATA_DISPLAY,
    isMain: false,
    groupBy: "data_display",
  },
  {
    title: "Feedback",
    name: "feedback",
    component: COMPONENTS.components.feedback,
    icon: NAVICONS.components.feedback,
    key: "",
    path: "/components/feedback/",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.FEEDBACK,
    isMain: true,
    groupBy: "feedback",
  },
  {
    title: "Dialog",
    name: "dialog",
    component: COMPONENTS.components.feedback.dialog,
    icon: "",
    key: "",
    path: "/components/feedback/dialog",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.FEEDBACK,
    isMain: false,
    groupBy: "feedback",
  },
  {
    title: "Progress",
    name: "progress",
    component: COMPONENTS.components.feedback.progress,
    icon: "",
    key: "",
    path: "/components/feedback/progress",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.FEEDBACK,
    isMain: false,
    groupBy: "feedback",
  },
  {
    title: "Snackbar",
    name: "snackbar",
    component: COMPONENTS.components.feedback.snackbar,
    icon: "",
    key: "",
    path: "/components/feedback/snackbar",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.FEEDBACK,
    isMain: false,
    groupBy: "feedback",
  },
  {
    title: "Surfaces",
    name: "surfaces",
    component: COMPONENTS.components.surfaces,
    icon: NAVICONS.components.surfaces,
    key: "",
    path: "/components/surface",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.SURFACES,
    isMain: true,
    groupBy: "surfaces",
  },
  {
    title: "Accordion",
    name: "accordion",
    component: COMPONENTS.components.surfaces.accordion,
    icon: "",
    key: "",
    path: "/components/surface/accordion",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.SURFACES,
    isMain: false,
    groupBy: "surfaces",
  },
  {
    title: "Card",
    name: "card",
    component: COMPONENTS.components.surfaces.card,
    icon: "",
    key: "",
    path: "/components/surface/card",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.SURFACES,
    isMain: false,
    groupBy: "surfaces",
  },
  {
    title: "Paper",
    name: "paper",
    component: COMPONENTS.components.surfaces.paper,
    icon: "",
    key: "",
    path: "/components/surface/paper",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.SURFACES,
    isMain: false,
    groupBy: "surfaces",
  },
  {
    title: "Navigation",
    name: "navigation",
    component: COMPONENTS.components.navigation,
    icon: NAVICONS.components.navigation,
    key: "",
    path: "/components/navigation/",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.NAVIGATION,
    isMain: true,
    groupBy: "navigation",
  },
  {
    title: "Menu",
    name: "menu",
    component: COMPONENTS.components.navigation.menu,
    icon: "",
    key: "",
    path: "/components/navigation/menu",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.NAVIGATION,
    isMain: false,
    groupBy: "navigation",
  },
  {
    title: "Pagination",
    name: "pagination",
    component: COMPONENTS.components.navigation.pagination,
    icon: "",
    key: "",
    path: "/components/navigation/pagination",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.NAVIGATION,
    isMain: false,
    groupBy: "navigation",
  },
  {
    title: "Tabs",
    name: "tabs",
    component: COMPONENTS.components.navigation.tabs,
    icon: "",
    key: "",
    path: "/components/navigation/tabs",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.NAVIGATION,
    isMain: false,
    groupBy: "navigation",
  },
  {
    title: "Layout",
    name: "layout",
    component: COMPONENTS.components.layout,
    icon: NAVICONS.components.layout,
    key: "",
    path: "/components/layout/",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.LAYOUT,
    isMain: true,
    groupBy: "layout",
  },
  {
    title: "Box",
    name: "box",
    component: COMPONENTS.components.layout.box,
    icon: "",
    key: "",
    path: "/components/layout/box",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.LAYOUT,
    isMain: false,
    groupBy: "layout",
  },
  {
    title: "Container",
    name: "container",
    component: COMPONENTS.components.layout.container,
    icon: "",
    key: "",
    path: "/components/layout/container",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.LAYOUT,
    isMain: false,
    groupBy: "layout",
  },
  {
    title: "Stack",
    name: "stack",
    component: COMPONENTS.components.layout.stack,
    icon: "",
    key: "",
    path: "/components/layout/stack",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.LAYOUT,
    isMain: false,
    groupBy: "layout",
  },
  {
    title: "MUI X",
    name: "mui_x",
    component: COMPONENTS.components.mui_x,
    icon: NAVICONS.components.mui_x,
    key: "",
    path: "/components/mui-x/",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.MUI_X,
    isMain: true,
    groupBy: "mui_x",
  },
  {
    title: "Data Grid",
    name: "data_grid",
    component: COMPONENTS.components.mui_x.data_grid,
    icon: "",
    key: "",
    path: "/components/mui-x/data-grid",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.MUI_X,
    isMain: false,
    groupBy: "mui_x",
  },
  {
    title: "Date/Time Picker",
    name: "date_time_picker",
    component: COMPONENTS.components.mui_x.date_time_picker,
    icon: "",
    key: "",
    path: "/components/mui-x/date-time-picker",
    section: SECTIONS.COMPONENTS,
    submenu: SUBMENU.MUI_X,
    isMain: false,
    groupBy: "mui_x",
  },
  {
    title: "Extras",
    name: "extras",
    component: COMPONENTS.third_party_libraries.extras,
    icon: NAVICONS.third_party_libraries.extras,
    key: "",
    path: "/third-party-libraries/extras/",
    section: SECTIONS.THIRD_PARTY_LIBRARIES,
    submenu: SUBMENU.EXTRAS,
    isMain: true,
    groupBy: "extras",
  },
  {
    title: "Chart",
    name: "chart",
    component: COMPONENTS.third_party_libraries.extras.chart,
    icon: "",
    key: "",
    path: "/3rd-party-libraries/extras/chart",
    section: SECTIONS.THIRD_PARTY_LIBRARIES,
    submenu: SUBMENU.EXTRAS,
    isMain: false,
    groupBy: "extras",
  },
  {
    title: "Rich Text",
    name: "rich_text",
    component: COMPONENTS.third_party_libraries.extras.rich_text,
    icon: "",
    key: "",
    path: "/3rd-party-libraries/extras/rich-text",
    section: SECTIONS.THIRD_PARTY_LIBRARIES,
    submenu: SUBMENU.EXTRAS,
    isMain: false,
    groupBy: "extras",
  },
  {
    title: "Upload File",
    name: "upload_file",
    component: COMPONENTS.third_party_libraries.extras.upload_file,
    icon: "",
    key: "",
    path: "/3rd-party-libraries/extras/upload-file",
    section: SECTIONS.THIRD_PARTY_LIBRARIES,
    submenu: SUBMENU.EXTRAS,
    isMain: false,
    groupBy: "extras",
  },
  {
    title: "Form Validation",
    name: "form_validation",
    component: COMPONENTS.third_party_libraries.extras.form_validation,
    icon: "",
    key: "",
    path: "/3rd-party-libraries/extras/form-validation",
    section: SECTIONS.THIRD_PARTY_LIBRARIES,
    submenu: SUBMENU.EXTRAS,
    isMain: false,
    groupBy: "extras",
  },
];
