// nms-creative-ui/src/components/Navigation/Tabs/Tabs/Tabs.jsx
import React from "react";
import { Tabs as NMSTabs } from "@mui/material";
import PropTypes from "prop-types";

const Tabs = ({
  children,
  component,
  indicatorColor,
  onChange,
  orientation,
  scrollButtons,
  slots,
  sx,
  textColor,
  value,
  variant,
  ...props
}) => {
  return (
    <NMSTabs
      component={component}
      indicatorColor={indicatorColor}
      onChange={onChange}
      orientation={orientation}
      scrollButtons={scrollButtons}
      slots={slots}
      sx={sx}
      textColor={textColor}
      value={value}
      variant={variant}
      {...props}
    >
      {children}
    </NMSTabs>
  );
};

Tabs.propTypes = {
  children: PropTypes.node,
  component: PropTypes.elementType,
  indicatorColor: PropTypes.oneOfType([
    PropTypes.oneOf(["primary", "secondary"]),
    PropTypes.string,
  ]),
  onChange: PropTypes.func,
  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
  scrollButtons: PropTypes.oneOfType([
    PropTypes.oneOf(["auto"]),
    PropTypes.bool,
  ]),
  slots: PropTypes.shape({
    EndScrollButtonIcon: PropTypes.elementType,
    StartScrollButtonIcon: PropTypes.elementType,
  }),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  textColor: PropTypes.oneOf(["inherit", "primary", "secondary"]),
  value: PropTypes.any,
  variant: PropTypes.oneOf(["fullWidth", "scrollable", "standard"]),
  ...NMSTabs.PropTypes,
};

Tabs.defaultProps = {
  children: "NMS Tabs",
  indicatorColor: "primary",
  orientation: "horizontal",
  textColor: "primary",
  variant: "standard",
};

export default Tabs;
