// nms-creative-ui/src/components/Navigation/Pagination/Pagination/Pagination.jsx
import React from "react";
import { Pagination as NMSPagination } from "@mui/material";
import PropTypes from "prop-types";

const Pagination = ({
  boundaryCount,
  color,
  count,
  defaultPage,
  onChange,
  page,
  renderItem,
  shape,
  size,
  sx,
  variant,
  ...props
}) => {
  return (
    <NMSPagination
      boundaryCount={boundaryCount}
      color={color}
      count={count}
      defaultPage={defaultPage}
      onChange={onChange}
      page={page}
      renderItem={renderItem}
      shape={shape}
      size={size}
      sx={sx}
      variant={variant}
      {...props}
    />
  );
};

Pagination.propTypes = {
  boundaryCount: PropTypes.number,
  color: PropTypes.oneOfType([
    PropTypes.oneOf(["primary", "secondary", "standard"]),
    PropTypes.string,
  ]),
  count: PropTypes.number,
  defaultPage: PropTypes.number,
  onChange: PropTypes.func,
  page: PropTypes.number,
  renderItem: PropTypes.func,
  shape: PropTypes.oneOf(["circular", "rounded"]),
  size: PropTypes.oneOfType([
    PropTypes.oneOf(["small", "medium", "large"]),
    PropTypes.string,
  ]),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["outlined", "text"]),
    PropTypes.string,
  ]),
  ...NMSPagination.PropTypes,
};

Pagination.defaultProps = {
  color: "standard",
  shape: "circular",
  size: "medium",
  variant: "text",
};

export default Pagination;
