// nms-creative-ui/src/pages/Components/MUI X/DataGrid/Preview/Custom.jsx
import React, { Fragment } from "react";
import { Chip, DataGrid, Grid, Icon } from "../../../../../components";
import * as SolarIconSet from "solar-icon-set";
import { GridActionsCellItem } from "@mui/x-data-grid";
import DesktopFilterComponent from "../../../../../components/Generic Components/Filter component/DesktopFilterComponent";

const columns = [
  {
    field: "avatar",
    headerName: "Avatar",
    width: 90,
    renderCell: (params) => (
      <img
        src={params.value}
        alt="Avatar"
        style={{ width: 40, height: 40, borderRadius: "50%" }}
      />
    ),
  },
  {
    field: "fullName",
    headerName: "Full Name",
    width: 150,
    editable: true,
  },
  {
    field: "personalEmail",
    headerName: "Personal Email",
    width: 150,
    editable: true,
  },

  {
    field: "category",
    headerName: "Category",
    width: 150,
    editable: true,
  },
  {
    field: "status",
    headerName: "Status",
    width: 150,
    editable: true,
    renderCell: (params) => {
      return (
        <Fragment>
          {params.value === "Active" ? (
            <Chip color="success" label="Active" />
          ) : (
            <Chip color="error" label="Inactive" />
          )}
        </Fragment>
      );
    },
  },
  {
    field: "actions",
    headerName: "Actions",
    width: 180,
    editable: false,
    cellClassName: "actions",
    renderCell: () => {
      return (
        <Fragment>
          <GridActionsCellItem
            icon={
              <Icon color="secondary">
                <SolarIconSet.Pen
                  iconStyle="BoldDuotone"
                  size={24}
                  label="Edit"
                />
              </Icon>
            }
          />
          <GridActionsCellItem
            icon={
              <Icon color="secondary">
                <SolarIconSet.Eye
                  iconStyle="BoldDuotone"
                  size={24}
                  label="Edit"
                />
              </Icon>
            }
          />
        </Fragment>
      );
    },
  },
];

const rows = [
  {
    id: 1,
    avatar: "/avatars/Avatar 1.svg",
    fullName: "Von Perez",
    personalEmail: "vp@gmail.com",
    category: "Office Based",
    status: "Active",
  },
  {
    id: 2,
    avatar: "/avatars/Avatar 2.svg",
    fullName: "Aldrin Villalobos",
    personalEmail: "vp@gmail.com",
    category: "Office Based",
    status: "Inactive",
  },
  {
    id: 3,
    avatar: "/avatars/Avatar 3.svg",
    fullName: "Elmer Bautista",
    personalEmail: "vp@gmail.com",
    category: "Office Based",
    status: "Active",
  },
  {
    id: 4,
    avatar: "/avatars/Avatar 4.svg",
    fullName: "Marjon Dupeng",
    personalEmail: "vp@gmail.com",
    category: "Office Based",
    status: "Inactive",
  },
];

const Custom = () => {
  return (
    <Grid container spacing={8}>
      <Grid item mobile={12} desktop={4}>
        <DesktopFilterComponent />
      </Grid>

      <Grid item mobile={12} desktop={8}>
        <DataGrid
          rows={rows}
          rowHeight={64}
          columns={columns}
          initialState={{
            pagination: {
              paginationModel: {
                pageSize: 5,
              },
            },
          }}
          pageSizeOptions={[5, 10, 25]}
          disableRowSelectionOnClick
          sx={{ border: 0 }}
        />
      </Grid>
    </Grid>
  );
};

export default Custom;
