// nms-creative-ui/src/components/Data Display/Chip/Chip.jsx
import React from "react";
import { Chip as NMSChip } from "@mui/material";
import PropTypes from "prop-types";

const Chip = ({
  avatar,
  children,
  color,
  component,
  deleteIcon,
  icon,
  label,
  onDelete,
  size,
  sx,
  variant,
  clickable,
  ...props
}) => {
  return (
    <NMSChip
      avatar={avatar}
      color={color}
      component={component}
      deleteIcon={deleteIcon}
      icon={icon}
      label={label}
      onDelete={onDelete}
      size={size}
      sx={sx}
      variant={variant}
      clickable = {clickable}
      {...props}
    />
  );
};

Chip.propTypes = {
  avatar: PropTypes.element,
  color: PropTypes.oneOfType([
    PropTypes.oneOf([
      "default",
      "primary",
      "secondary",
      "error",
      "info",
      "success",
      "warning",
    ]),
    PropTypes.string,
  ]),
  component: PropTypes.elementType,
  deleteIcon: PropTypes.element,
  icon: PropTypes.element,
  label: PropTypes.node,
  onDelete: PropTypes.func,
  size: PropTypes.oneOfType([
    PropTypes.oneOf(["medium", "small"]),
    PropTypes.string,
  ]),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["filled", "outlined"]),
    PropTypes.string,
  ]),
  ...NMSChip.PropTypes,
};

Chip.defaultProps = {
  color: "primary",
  size: "medium",
  variant: "filled",
};

export default Chip;
