// nms-creative-ui/src/components/Navigation/Menu/Menu/Menu.jsx
import React from "react";
import { Menu as NMSMenu } from "@mui/material";
import PropTypes from "prop-types";

const Menu = ({
  open,
  anchorEl,
  children,
  MenuListProps,
  onClose,
  sx,
  transitionDuration,
  TransitionProps,
  variant,
  ...props
}) => {
  return (
    <NMSMenu
      open={open}
      anchorEl={anchorEl}
      MenuListProps={MenuListProps}
      onClose={onClose}
      sx={sx}
      transitionDuration={transitionDuration}
      TransitionProps={TransitionProps}
      variant={variant}
      {...props}
    >
      {children}
    </NMSMenu>
  );
};

Menu.propTypes = {
  open: PropTypes.bool,
  anchorEl: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  children: PropTypes.node,
  MenuListProps: PropTypes.object,
  onClose: PropTypes.func,
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  transitionDuration: PropTypes.oneOfType([
    PropTypes.oneOf(["auto"]),
    PropTypes.number,
    PropTypes.shape({
      appear: PropTypes.number,
      enter: PropTypes.number,
      exit: PropTypes.number,
    }),
  ]),
  TransitionProps: PropTypes.object,
  variant: PropTypes.oneOf(["menu", "selectedMenu"]),
  ...NMSMenu.PropTypes,
};

Menu.defaultProps = {
  children: "NMS Menu",
  transitionDuration: "auto",
  variant: "selectedMenu",
};

export default Menu;
