// nms-creative-ui/rollup.config.js
// rollup.config.js
import { defineConfig } from "rollup";
import resolve from "@rollup/plugin-node-resolve";
import commonjs from "@rollup/plugin-commonjs";
import babel from "@rollup/plugin-babel";
import copy from "rollup-plugin-copy";
import alias from "@rollup/plugin-alias";
import postcss from "rollup-plugin-postcss";
import external from "rollup-plugin-peer-deps-external";
import terser from "@rollup/plugin-terser";

export default defineConfig({
  input: "src/index.js",
  output: [
    { file: "dist/index.js", format: "cjs" },
    { file: "dist/index.html" },
    {
      file: "dist/index.es.js",
      format: "es",
      exports: "named",
      sourcemap: true,
    },
  ],
  plugins: [
    postcss({
      plugins: [],
      minimize: true,
    }),
    resolve({
      extensions: [".js", ".jsx"], // Add .jsx extension
    }),
    commonjs(),
    babel({
      exclude: "node_modules/**",
      presets: ["@babel/env", "@babel/preset-react"],
      babelHelpers: "bundled",
    }),
    copy({
      targets: [
        {
          src: "src/fonts/*",
          dest: "dist/fonts",
        },
      ],
    }),
    alias({
      entries: [{ find: "./src/", replacement: /^nms-creative-ui-hey-you\// }],
    }),
    external(),
    terser(),
  ],
  external: [
    "react",
    "react-dom",
  ],
});
