// nms-creative-ui/src/components/Navigation/Pagination/TablePagination/TablePagination.jsx
import React from "react";
import { TablePagination as NMSTablePagination } from "@mui/material";
import PropTypes from "prop-types";
import TablePaginationActions from "@mui/material/TablePagination/TablePaginationActions";

const TablePagination = (props) => {
  const {
    count,
    onPageChange,
    page,
    rowsPerPage,
    component,
    onRowsPerPageChange,
    rowsPerPageOptions,
    slots,
    sx,
  } = props;
  return (
    <NMSTablePagination
      count={count}
      onPageChange={onPageChange}
      page={page}
      rowsPerPage={rowsPerPage}
      component={component}
      onRowsPerPageChange={onRowsPerPageChange}
      rowsPerPageOptions={rowsPerPageOptions}
      slots={slots}
      sx={sx}
    />
  );
};

TablePagination.propTypes = {
  count: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  rowsPerPage: PropTypes.number.isRequired,
  component: PropTypes.elementType,
  onRowsPerPageChange: PropTypes.func,
  rowsPerPageOptions: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
      }),
    ])
  ),
  slots: PropTypes.shape({
    firstButton: PropTypes.elementType,
    firstButtonIcon: PropTypes.elementType,
    lastButton: PropTypes.elementType,
    lastButtonIcon: PropTypes.elementType,
    nextButton: PropTypes.elementType,
    nextButtonIcon: PropTypes.elementType,
    previousButton: PropTypes.elementType,
    previousButtonIcon: PropTypes.elementType,
  }),
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  ...NMSTablePagination.PropTypes,
};

TablePagination.defaultProps = {
  labelRowsPerPage: "Page Hehehe:",
  rowsPerPageOptions: [10, 25, 50, 100],
  showFirstButton: false,
  showLastButton: false,
};

export default TablePagination;
