// nms-creative-ui/src/pages/Getting Started/Foundation Page/Colors/ColorsPage.jsx
import { Box, Stack, Typography } from "@mui/material";
import React from "react";
import * as SolarIconSet from "solar-icon-set";
import { Grid, Icon, IconButton } from "../../../../components";
import {
  DANGER_COLORS,
  INFO_COLORS,
  NEUTRAL_COLORS,
  PRIMARY_COLORS,
  SECONDARY_COLORS,
  SUCCESS_COLORS,
  WARNING_COLORS,
} from "./data/colors";
import CopyToClipboard from "react-copy-to-clipboard";
import { useSnackbar } from "notistack";
import CustPaper from "../../../../components/Generic Components/CustPaper/CustPaper";


const ColorsPage = () => {
  const { enqueueSnackbar, closeSnackbar } = useSnackbar();

  const handleCopyToClipboard = (color) => {
    enqueueSnackbar(`Copied: ${color}`, {
      variant: "success",
      autoHideDuration: 3000,

      action: (key) => (
        <IconButton
          color="success"
          size="medium"
          aria-label="close"
          severity="success"
          onClick={() => closeSnackbar(key)}
        >
          <SolarIconSet.CloseCircle fontSize={24} />
        </IconButton>
      ),
      icon: (
        <Icon color="#59C227">
          <SolarIconSet.InfoSquare size={24} iconStyle="BoldDuotone" />
        </Icon>
      ),
      style: {
        backgroundColor: "#EFFDD6",
        color: "#59C227",
      },
    });
  };

  return (
    <Stack spacing={6}>
      <Box>
        <Stack direction={"column"} spacing={3}>
          <Stack direction={"row"} spacing={1}>
            <Icon color="secondary">
              <SolarIconSet.BoxMinimalistic size={24} iconStyle="BoldDuotone" />
            </Icon>

            <Typography variant="body1">Foundation</Typography>
          </Stack>

          <Stack direction={"column"} spacing={1}>
            <Typography variant="h2">Colors</Typography>

            <Typography variant="body1">
              Express meaning with colors. Explore a spectrum of colors designed
              to evoke specific emotions and guide user experience.
            </Typography>
          </Stack>
          <Stack direction={"column"} spacing={1}>
            <a
              href="https://mui.com/material-ui/customization/color/"
              target="_blank"
            >
              <Typography>
                https://mui.com/material-ui/customization/color/
              </Typography>
            </a>

            <a href="https://colors.eva.design" target="_blank">
              <Typography>https://colors.eva.design</Typography>
            </a>
          </Stack>
        </Stack>
      </Box>
      <Grid container spacing={4}>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper>
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Primary</Typography>
            </Box>

            {PRIMARY_COLORS.map((color) => (
              <Stack direction="row">
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Secondary</Typography>
            </Box>

            {SECONDARY_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Info</Typography>
            </Box>

            {INFO_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Success</Typography>
            </Box>

            {SUCCESS_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Warning</Typography>
            </Box>

            {WARNING_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Danger</Typography>
            </Box>

            {DANGER_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
        <Grid item mobile={12} desktop={4} tablet={4}>
          <CustPaper
            elevation={4}
            sx={{ display: "flex", flexDirection: "column", gap: "8px" }}
          >
            <Box sx={{ paddingBottom: "24px" }}>
              <Typography variant="h4">Neutral</Typography>
            </Box>

            {NEUTRAL_COLORS.map((color) => (
              <Stack direction="row" spacing={2}>
                <CopyToClipboard
                  key={color.id}
                  text={color.color_hex}
                  onCopy={() => handleCopyToClipboard(color.color_hex)}
                >
                  <Stack
                    direction="row"
                    spacing={2}
                    sx={{ paddingBottom: "8px" }}
                  >
                    <Box
                      key={color.id}
                      sx={{
                        width: "50px",
                        height: "50px",
                        borderRadius: "10px",
                        backgroundColor: color.color_hex,
                      }}
                    />
                    <Stack>
                      <Typography variant="h6">{color.color_name}</Typography>
                      <Typography variant="body1">{color.color_hex}</Typography>
                    </Stack>
                  </Stack>
                </CopyToClipboard>
              </Stack>
            ))}
          </CustPaper>
        </Grid>
      </Grid>
    </Stack>
  );
};

export default ColorsPage;
