// nms-creative-ui/src/components/Input/Button/Button/Button.jsx
import React from "react";
import { Button as NMSButton } from "@mui/material";
import PropTypes from "prop-types";

const Button = ({
  children,
  color,
  component,
  endIcon,
  href,
  size,
  startIcon,
  sx,
  variant,
  ...props
}) => {
  return (
    <NMSButton
      color={color}
      component={component}
      endIcon={endIcon}
      href={href}
      size={size}
      startIcon={startIcon}
      sx={sx}
      variant={variant}
      {...props}
    >
      {children}
    </NMSButton>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  color: PropTypes.oneOfType([
    PropTypes.oneOf([
      "inherit",
      "default",
      "primary",
      "secondary",
      "error",
      "info",
      "success",
      "warning",
    ]),
    PropTypes.string,
  ]),
  component: PropTypes.elementType,
  endIcon: PropTypes.node,
  href: PropTypes.string,
  size: PropTypes.oneOfType([
    PropTypes.oneOf(["small", "medium", "large"]),
    PropTypes.string,
  ]),
  startIcon: PropTypes.node,
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["contained", "outlined", "text"]),
    PropTypes.string,
  ]),
  ...NMSButton.PropTypes,
};

Button.defaultProps = {
  children: "NMS Button",
  color: "primary",
  size: "large",
  variant: "contained",
};

export default Button;
