// nms-creative-ui/src/pages/Getting Started/InstallationPage.jsx
import * as SolarIconSet from "solar-icon-set";
import React, { useState } from "react";
import {
  Box,
  Grid,
  Icon,
  IconButton,
  Paper,
  Stack,
  Typography,
} from "../../components";
import { Fragment } from "react";
import { useSnackbar } from "notistack";
import CopyToClipboard from "react-copy-to-clipboard";

const InstallationPage = () => {
  const [copied, setCopied] = useState(false);
  const { enqueueSnackbar, closeSnackbar } = useSnackbar();

  const importSnippet = `import '@fontsource/lato/100.css';
  import '@fontsource/lato/300.css';
  import '@fontsource/lato/400.css';
  import '@fontsource/lato/700.css';
  import '@fontsource/lato/900.css';`;

  const cdnSnippet = `ows`

  const handleCopy = () => {
    setCopied(true);
  };

  const handleCopyToClipboard = () => {
    enqueueSnackbar("Copied to Clipboard!", {
      variant: "info",
      autoHideDuration: 3000,

      action: (key) => (
        <IconButton
          color="info"
          size="medium"
          aria-label="close"
          severity="info"
          onClick={() => closeSnackbar(key)}
        >
          <SolarIconSet.CloseCircle fontSize={24} />
        </IconButton>
      ),
      icon: (
        <Icon color="success">
          <SolarIconSet.InfoSquare size={24} iconStyle="BoldDuotone" />
        </Icon>
      ),
      style: {
        backgroundColor: "#CBFEFA",
        color: "#0090CB",
      },
    });
  };

  return (
    <Grid>
      <Stack direction={"column"} spacing={6}>
        <Grid item mobile={12} desktop={12}>
          <Stack direction="column" spacing={3}>
            <Typography variant="h2">Installation</Typography>
            <Typography variant="body1">
              Deploy the latest customized React UI framework, NMS Creative UI,
              built upon the foundation of Material UI.
            </Typography>
          </Stack>
        </Grid>

        <Grid item mobile={12} desktop={12}>
          <Stack direction="column" spacing={3}>
            <Typography variant="h3">Requirements</Typography>
            <ul>
              <li>
                <Typography variant="body1">Node.js v16.14 +</Typography>
              </li>
              <li>
                <Typography variant="body1">React 18+</Typography>
              </li>
              <li>
                <Typography variant="body1">
                  Do not delete the any file in (package-lock.json / yarn.lock)
                  the folder.
                </Typography>
              </li>
            </ul>
          </Stack>
        </Grid>

        <Stack direction={"column"} spacing={3}>
          <Stack direction={"column"} spacing={1}>
            <Typography variant="h3">Default Installation</Typography>
            <Typography variant="body1">
              Run one of the following commands to add NMS Creative UI to your
              project:
            </Typography>
          </Stack>
          <Paper
            elevation={4}
            sx={{ "&.MuiPaper-elevation4": { padding: "24px" } }}
          >
            <Stack
              direction={"row"}
              justifyContent={"space-between"}
              alignItems={"center"}
            >
              <Typography>npm install nms-creative-ui</Typography>
              <CopyToClipboard
                text={`npm install nms-creative-ui`}
                onCopy={handleCopyToClipboard}
              >
                <IconButton>
                  <Icon
                    size={24}
                    solariconset={SolarIconSet.Copy}
                    iconStyle={"BoldDuotone"}
                    color={"inherit"}
                  />
                </IconButton>
              </CopyToClipboard>
            </Stack>
          </Paper>
        </Stack>

        <Box>
          <Stack direction={"column"} spacing={3}>
            <Stack direction={"column"} spacing={1}>
              <Typography variant="h3">Icons</Typography>
              <Typography variant="body1">
                NMS Creative UI uses the Solar Icon Set. Add it to your project
                via npm.
              </Typography>
            </Stack>
            <Paper
              elevation={4}
              sx={{ "&.MuiPaper-elevation4": { padding: "24px" } }}
            >
              <Stack
                direction={"row"}
                justifyContent={"space-between"}
                alignItems={"center"}
              >
                <Typography>npm install solar-icon-set</Typography>
                <CopyToClipboard
                  text={`npm install solar-icon-set`}
                  onCopy={handleCopyToClipboard}
                >
                  <IconButton>
                    <Icon
                      size={24}
                      solariconset={SolarIconSet.Copy}
                      iconStyle={"BoldDuotone"}
                      color={"inherit"}
                    />
                  </IconButton>
                </CopyToClipboard>
              </Stack>
            </Paper>
          </Stack>
        </Box>

        <Box>
          <Stack direction={"column"} spacing={3}>
            <Typography variant="h3">Lato Font</Typography>
            <Typography variant="body1">
              NMS Creative UI employs the Lato font as the default choice.
              Integrate it into your project using Fontsource or through the
              Google Fonts CDN. If the Lato font is not working, please follow
              one of these steps.
            </Typography>

            <Typography variant="body1">
              Add it to your project via Fontsource:
            </Typography>
            <Paper
              elevation={4}
              sx={{ "&.MuiPaper-elevation4": { padding: "24px" } }}
            >
              <Stack
                direction={"row"}
                justifyContent={"space-between"}
                alignItems={"center"}
              >
                <Typography>npm install @fontsource/lato</Typography>
                <CopyToClipboard
                  text={`npm install @fontsource/lato`}
                  onCopy={handleCopyToClipboard}
                >
                  <IconButton>
                    <Icon
                      size={24}
                      solariconset={SolarIconSet.Copy}
                      iconStyle={"BoldDuotone"}
                      color={"inherit"}
                    />
                  </IconButton>
                </CopyToClipboard>
              </Stack>
            </Paper>

            <Typography variant="body1">
              Then you can import it in your entry point like this:
            </Typography>
            <Paper
              elevation={4}
              sx={{ "&.MuiPaper-elevation4": { padding: "24px" } }}
            >
              <Stack
                direction={"row"}
                justifyContent={"space-between"}
                alignItems={"start"}
              >
                <Typography sx={{ whiteSpace: "pre-line" }}>
                  {importSnippet}
                </Typography>
                <CopyToClipboard
                  text={importSnippet}
                  onCopy={handleCopyToClipboard}
                >
                  <IconButton>
                    <Icon
                      size={24}
                      solariconset={SolarIconSet.Copy}
                      iconStyle={"BoldDuotone"}
                      color={"inherit"}
                    />
                  </IconButton>
                </CopyToClipboard>
              </Stack>
            </Paper>

            <Typography variant="body1">
            Or using Google Web Fonts CDN:
            </Typography>
            <Paper
              elevation={4}
              sx={{"&.MuiPaper-elevation4": { padding: "24px" }  }}
            >
              <Stack
                direction={"row"}
                justifyContent={"space-between"}
                alignItems={"start"}
              >
                <Typography>
                {cdnSnippet} 
                </Typography>
                <CopyToClipboard
                  text={importSnippet}
                  onCopy={handleCopyToClipboard}
                >
                  <IconButton>
                    <Icon
                      size={24}
                      solariconset={SolarIconSet.Copy}
                      iconStyle={"BoldDuotone"}
                      color={"inherit"}
                    />
                  </IconButton>
                </CopyToClipboard>
              </Stack>
            </Paper>

           
          </Stack>
        </Box>
      </Stack>
    </Grid>
  );
};

export default InstallationPage;
