// nms-creative-ui/src/components/Generic Components/AppBar Component/AppBar.jsx
import { Stack, Toolbar, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { CustAppBar } from "../General Components/CustAppBar";
import { Link } from "react-router-dom";
import * as SolarIconSet from "solar-icon-set";
import IconButton from "../../Input/Button/IconButton/IconButton";
import Button from "../../Input/Button/Button";


const AppBar = ({ handleDrawerToggle, darkMode, toggleDarkMode }) => {
  return (
    <CustAppBar
    position="fixed"
    color="inherit"
    sx={{
      "&.MuiPaper-elevation4": {
        padding: 0,
        boxShadow: "none",
      },
    }}
  >
     <Toolbar
        sx={{
          justifyContent: { mobile: "space-between", laptop: "flex-end" },
          paddingTop: "24px",
          paddingBottom: "24px",
          paddingLeft: "48px",
          paddingRight: "48px",
        }}
      >
        <IconButton
          aria-label="open drawer"
          edge="start"
          onClick={handleDrawerToggle} // Use handleDrawerToggle function here
          sx={{display: { laptop: "none" } }}
        >
          <SolarIconSet.List2 size={32} iconStyle="BoldDuotone" />
        </IconButton>
        <Stack direction={"row"} spacing={1} alignItems={"center"}>
          <Link to={"/"}>
            <Button variant={"text"}>Home</Button>
          </Link>
          <IconButton onClick={toggleDarkMode}>
            {darkMode ? (
              <SolarIconSet.Sun
                iconStyle="BoldDuotone"
                size={24}
                color="#FF4545"
              />
            ) : (
              <SolarIconSet.MoonStars
                iconStyle="BoldDuotone"
                size={24}
                color="#FF4545"
              />
            )}
          </IconButton>
        </Stack>
      </Toolbar>
    </CustAppBar>
  )
};

export default AppBar;
