// nms-creative-ui/src/components/Generic Components/AppBar Component/AppBar2.jsx
import { AppBar, IconButton, Toolbar, Typography } from "@mui/material";
import React from "react";
import * as SolarIconSet from "solar-icon-set";
import Stack from "../../Layout/Stack";
import { Link } from "react-router-dom";
import Button from "../../Input/Button/Button";
import { CustAppBar } from "../General Components/CustAppBar";



const AppBar2 = ({ darkMode, toggleDarkMode }) => {

  const handleDrawerToggle = () => {
    if (!isClosing) {
      setMobileOpen(!mobileOpen);
    }
  };
  
  return (
    <CustAppBar
    position="fixed"
    color="inherit"
    sx={{
      "&.MuiPaper-elevation4": {
        padding: 0,
        boxShadow: "none",
      },
    }}
  >
     <Toolbar
        sx={{
          justifyContent: { mobile: "space-between", laptop: "flex-end" },
          paddingTop: "24px",
          paddingBottom: "24px",
          paddingLeft: "48px",
          paddingRight: "48px",
        }}
      >
        <IconButton
          aria-label="open drawer"
          edge="start"
          onClick={handleDrawerToggle} // Use handleDrawerToggle function here
          sx={{display: { laptop: "none" } }}
        >
          <SolarIconSet.List2 size={32} iconStyle="BoldDuotone" />
        </IconButton>
        <Stack direction={"row"} spacing={1} alignItems={"center"}>
          <Link to={"/"}>
            <Button variant={"text"}>Home</Button>
          </Link>
          <IconButton onClick={toggleDarkMode}>
            {darkMode ? (
              <SolarIconSet.Sun
                iconStyle="BoldDuotone"
                size={24}
                color="#FF4545"
              />
            ) : (
              <SolarIconSet.MoonStars
                iconStyle="BoldDuotone"
                size={24}
                color="#FF4545"
              />
            )}
          </IconButton>
        </Stack>
      </Toolbar>
    </CustAppBar>
  );
};

export default AppBar2;
