// nms-creative-ui/src/pages/Components/Surfaces/Card/JSX/BannerCardSample.jsx
export const BannerCardSample = `import React from "react";
import {
  Avatar,
  Box,
  Card,
  CardContent,
  Grid,
  Stack,
  Typography
} from "nms-creative-ui";

const DATA = [
  {
    id: 0,
    bgImage: "/banners/morningBanner.png",
    avatarSrc: "/avatars/Avatar 9.png",
    desc: "Good Morning",
    user: "John Doe!",
  },
  {
    id: 1,
    bgImage: "/banners/afternoonBanner.png",
    avatarSrc: "/avatars/Avatar 9.png",
    desc: "Good Afternoon",
    user: "John Doe!",
  },
  {
    id: 2,
    bgImage: "/banners/eveningBanner.png",
    avatarSrc: "/avatars/Avatar 9.png",
    desc: "Good Evening",
    user: "John Doe!",
  },
];

const BannerCard = () => {
  return (
    <Grid container spacing={2}>
      {DATA.map((item, index) => (
        <Grid key={index} item mobile={12}>
          <Card
            sx={{
              backgroundSize: "cover",
              backgroundImage: <>url(${item.bgImage})</>,
              backgroundPosition: "center",
              padding: "24px",
            }}
          >
            <Box sx={{ padding: "24px" }}>
              <CardContent>
                <Stack direction="row" spacing={4} flexWrap="wrap">
                  <Avatar
                    alt="User Avatar"
                    src={item.avatarSrc}
                    sx={{ width: "100px", height: "100px" }}
                  />
                  <Box sx={{ display: "flex", alignItems: "center" }}>
                    <Stack direction="column">
                      <Typography color="#FEFEFE" variant="h3">
                        {item.desc}
                      </Typography>
                      <Typography color="#FEFEFE" variant="h3">
                        {item.user}
                      </Typography>
                    </Stack>
                  </Box>
                </Stack>
              </CardContent>
            </Box>
          </Card>
        </Grid>
      ))}
    </Grid>
  );
};

export default BannerCard;
`;
