// nms-creative-ui/src/components/Input/ToggleButton/ToggleButtonGroup/ToggleButtonGroup.jsx
import React from "react";
import PropTypes from "prop-types";
import { ToggleButtonGroup as NMSToggleButtonGroup } from "@mui/material";

const ToggleButtonGroup = ({
  children,
  exclusive,
  onChange,
  orientation,
  size,
  fullWidth,
  sx,
  value,
  ...props
}) => {
  return (
    <NMSToggleButtonGroup
      exclusive={exclusive}
      onChange={onChange}
      orientation={orientation}
      size={size}
      fullWidth={fullWidth}
      sx={sx}
      value={value}
      {...props}
    >
      {children}
    </NMSToggleButtonGroup>
  );
};

ToggleButtonGroup.propTypes = {
  children: PropTypes.node,
  exclusive: PropTypes.bool,
  onChange: PropTypes.func,
  fullWidth: PropTypes.bool,
  orientation: PropTypes.oneOf(["horizontal", "vertical"]),
  size: PropTypes.oneOf(["small", "medium", "large"]),
  sx: PropTypes.object,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
};

ToggleButtonGroup.defaultProps = {
  size: "large",
  freeSolo: false,
  fullWidth: true,
};

export default ToggleButtonGroup;
