// nms-creative-ui/src/components/Data Display/Avatar/Avatar/Avatar.jsx
import React from "react";
import { Avatar as NMSAvatar } from "@mui/material";
import PropTypes from "prop-types";

const Avatar = ({
  alt,
  children,
  component,
  imgProps,
  size,
  src,
  sx,
  variant,
  ...props
}) => {
  return (
    <NMSAvatar
      alt={alt}
      component={component}
      imgProps={imgProps}
      size={size}
      src={src}
      sx={sx}
      variant={variant}
      {...props}
    >
      {children}
    </NMSAvatar>
  );
};

Avatar.propTypes = {
  alt: PropTypes.string,
  children: PropTypes.node,
  component: PropTypes.elementType,
  imgProps: PropTypes.object,
  size: PropTypes.string,
  src: PropTypes.string,
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])
    ),
    PropTypes.func,
    PropTypes.object,
    PropTypes.bool,
  ]),
  variant: PropTypes.oneOfType([
    PropTypes.oneOf(["circular", "rounded", "square"]),
    PropTypes.string,
  ]),
};

Avatar.defaultProps = {
  children: "NMS",
  variant: "circular",
};

export default Avatar;
