// nms-creative-ui/src/components/Input/Switch/Switch.jsx
import React, { useState } from "react";
import { FormControlLabel, Switch as NMSSwitch } from "@mui/material";
import { PropTypes } from "prop-types";

const Switch = ({
  color,
  component,
  disabled,
  endIcon,
  fullWidth,
  size,
  startIcon,
  sx,
  variant,
  label,
  labelPlacement,
  checked,
  onChange,
}) => {
  return (
    <FormControlLabel
      control={
        <NMSSwitch
          checked={checked}
          color={color}
          component={component}
          disabled={disabled}
          endIcon={endIcon}
          fullWidth={fullWidth}
          size={size}
          startIcon={startIcon}
          sx={sx}
          variant={variant}
          onChange={onChange}
        />
      }
      label={label}
      labelPlacement={labelPlacement}
    />
  );
};

Switch.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  component: PropTypes.elementType,
  endIcon: PropTypes.node,
  fullWidth: PropTypes.bool,
  size: PropTypes.string,
  startIcon: PropTypes.node,
  sx: PropTypes.arrayOf(PropTypes.func),
  variant: PropTypes.string,
  label: PropTypes.string,
  labelPlacement: PropTypes.string,
};

Switch.defaultProps = {
  color: "primary",
  size: "medium",
};

export default Switch;
