// nms-creative-ui/src/pages/Components/Data Display/Badge/Preview/Status.jsx
import React, { Fragment } from "react";
import { Avatar, Badge, Icon, Stack } from "../../../../../components";
import * as SolarIconSet from "solar-icon-set";
import { styled } from "@mui/material";

const Status = () => {
  const StyledBadge = styled(Badge)(({ theme }) => ({
    "& .MuiBadge-badge": {
      backgroundColor: "#FF4545",
      color: "#FF4545",
      boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
      "&::after": {
        position: "absolute",
        top: 0,
        left: 0,
        width: "100%",
        height: "100%",
        borderRadius: "50%",
        animation: "ripple 1.2s infinite ease-in-out",
        border: "1px solid currentColor",
        content: '""',
      },
    },
    "@keyframes ripple": {
      "0%": {
        transform: "scale(.8)",
        opacity: 1,
      },
      "100%": {
        transform: "scale(2.4)",
        opacity: 0,
      },
    },
  }));

  return (
    <Fragment>
      <StyledBadge
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "right",
        }}
        variant="dot"
        color={"primary"}
        overlap="circular"
      >
        <Avatar />
      </StyledBadge>

      <Badge
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "right",
        }}
        variant="dot"
        color={"primary"}
        overlap="circular"
        badgeContent={
          <Icon size={24} solariconset={SolarIconSet.ClockCircle} />
        }
      >
        <Avatar />
      </Badge>

      <Badge
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "right",
        }}
        variant="dot"
        color={"primary"}
        overlap="circular"
      >
        <Avatar />
      </Badge>
    </Fragment>
  );
};

export default Status;
